src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/globalmodule.h

```diff
@@ -33,6 +40,48 @@ class ASPNET_CORE_GLOBAL_MODULE : NonCopyable, public CGlobalModule
         _In_ IGlobalConfigurationChangeProvider * pProvider
     ) override;
 
+    GLOBAL_NOTIFICATION_STATUS
+    OnGlobalApplicationStop(
+        IN IHttpApplicationStopProvider* pProvider
+    ) override;
+
 private:
     std::shared_ptr<APPLICATION_MANAGER> m_pApplicationManager;
+    std::thread m_shutdown;
+
+    void StartShutdown()
+    {
+        // Shutdown has already been started
+        if (m_shutdown.joinable())
```

Comment:
Does this stay true once the thread completes?

Reply:
It will be true if the thread is still running or if the thread completed but hasn't had `.join()` called on it.

Reply:
Yeah, both places `StartShutdown` is called check `g_fInShutdown` which is set from the thread in `StartShutdown`. I think putting a check for `g_fInShutdown` in `StartShutdown` would be a good idea though even if it's redundant.